src/coreclr/jit/flowgraph.cpp

```diff
@@ -88,11 +88,11 @@ PhaseStatus Compiler::fgInsertGCPolls()
     }
 #endif // DEBUG
 
-    BasicBlock* block;
-
     // Walk through the blocks and hunt for a block that needs a GC Poll
-    for (block = fgFirstBB; block != nullptr; block = block->bbNext)
+    for (BasicBlock* block : Blocks())
     {
+        compCurBB = block;
```

Comment:
I should reset this at the end of this method so other places doesn't falsely depend on the last block set in `compCurBB`.

Reply:
Are you asking if you should reset this after we finish iterating over the blocks?

It might be nice (in the future) to set this to null after each loop that uses it.
But that would be a risky chnage to make as itr is possible that the next phase (or some next phase) might be using compCurBB without setting it (which is what this method was doing).

Reply:
I realized that I should do it. While debugging this issue I remember saying a case where we were just taking the last BB `compCurBB`  for processing and that made me think that we should do it properly here. To be safe, we can do it after the loop before returning. We might get AV with that which will help us discover more situation where we use unintended value of `compCurBB`.